src/api/vdom.js

```diff
@@ -295,6 +295,21 @@ function newElementVoid (tag, ...args) {
 // Text override ensures their calls can queue if using function helpers.
 const newText = wrapIdomFunc(text);
 
+export function child (ch) {
+  const ctype = typeof ch;
+  if (ctype === 'function') {
+    ch();
+  } else if (ctype === 'string' || ctype === 'number') {
+    newText(ch);
+  } else if (Array.isArray(ch)) {
+    ch.forEach(child);
+  } else if (ctype === 'object' && ch.$$typeof) {
+    const { key, props, ref, type } = ch;
```

Comment:
`preact-compat` does support `$$typeof` the same as React, yup.  Preact itself currently relies on duck type checking of VNodes, and the format for VNodes is different - it uses the DOM names (`{ nodeName:'', attributes:{}, children:[] }`).  That may change in the future, since we'll likely drop the `VNode` constructor entirely and go to POJOs, in which case it might be useful to add a `$$typeof` flag.  It might also be worth just switching to `{ type, props, children }` as React uses, since `props` is used as a term basically everywhere else.